src/Pos/Types/Block/Functions.hs

```diff
@@ -496,6 +509,47 @@ verifyBlock VerifyBlockParams {..} blk =
               , "Some of block's PSKs have the same issuer, which is prohibited")
             ]
         | otherwise = mempty
+    verifyVersions bv = case blk of
+        Right mainBlk | bv >= mainBlk ^. gbHeader . gbhExtra . mehBlockVersion ->
```

Comment:
Here should be condition:
`lastKnownBlockVersion >= min (lastAdoptedBV, mainBlk ^. gbHeader . gbhExtra . mehBlockVersion)`